src/contracts/libraries/BN254.sol

```diff
@@ -127,6 +131,49 @@ library BN254 {
         require(success, "ec-add-failed");
     }
 
+    /**
+     * @notice an optimized ecMul implementation that takes O(log_2(s)) ecAdds
+     * @param p the point to multiply
+     * @param s the scalar to multiply by
+     * @dev this function is only safe to use if the scalar is 9 bits or less
+     */ 
+    function scalar_mul_tiny(BN254.G1Point memory p, uint16 s) internal view returns (BN254.G1Point memory) {
```

Comment:
function makes sense -- do we have test coverage ensuring equivalence with ecMul ?
if not, can you please file an issue for adding this test coverage?